File: library/src/main/java/com/sum/library/domain/BaseRepository.java
package com.sum.library.domain;

import android.arch.lifecycle.MutableLiveData;

/**
 * Created by sdl on 2019/2/18.
 */
public class BaseRepository {

    private MutableLiveData<ActionState> mActionLiveData;

    public BaseRepository() {
        mActionLiveData = new MutableLiveData<>();
    }

    //发送界面操作状态
    public void sendActionState(int state) {
        sendActionState(ActionState.obtain(state));
    }

    public void sendActionState(int state, String msg) {
        ActionState action = ActionState.obtain(state);
        action.setMsg(msg);
        sendActionState(action);
    }

    private void sendActionState(ActionState state) {
        mActionLiveData.postValue(state);
    }

    //viewModel发送界面操作状态
    MutableLiveData<ActionState> registerActionState(ActionState state) {
        mActionLiveData.postValue(state);
        return mActionLiveData;
    }
}

File: library/src/main/java/com/sum/library/domain/BaseRepository.kt
package com.sum.library.domain

import android.arch.lifecycle.LiveData
import android.arch.lifecycle.MutableLiveData
import com.sum.library.net.Retrofit2Helper
import retrofit2.Retrofit

/**
 * Created by sdl on 2018/7/13.
 */
open class BaseRepository {

    //界面操作状态(私有操作,避免直接访问)
    private val mActionLiveData by lazy {
        MutableLiveData<ActionState>()
    }

    protected val mRetrofit: Retrofit by lazy {
        Retrofit2Helper.getRetrofit()
    }

    //发送界面操作状态
    fun sendActionState(state: Int) {
        sendActionState(ActionState.obtain(state))
    }

    fun sendActionState(state: Int, msg: String) {
        val action = ActionState.obtain(state)
        action.msg = msg
        sendActionState(action)
    }

    fun sendActionState(state: ActionState) {
        mActionLiveData.postValue(state)
    }

    //viewModel发送界面操作状态
    fun registerActionState(state: ActionState): LiveData<ActionState> {
        mActionLiveData.postValue(state)
        return mActionLiveData
    }
}